pytorch_lightning/plugins/pipe_plugin.py

```diff
@@ -0,0 +1,356 @@
+    model.trainer.optimizers = optimizers
+    model.trainer.lr_schedulers = lr_schedulers
+    model.trainer.optimizer_frequencies = optimizer_frequencies
+
+
+def do_nothing_optimizer_closure():
+    return
+
+
+def cleanup(ctx, model):
+    del model
+
+
+def run_optimizer(ctx, model):
+    trainer = model.trainer
+    model_ref = trainer.get_model()
+    opt_idx = ctx["opt_idx"]
+    args = ctx["args"]
+    kwargs = ctx["kwargs"]
+    batch_idx = ctx["batch_idx"]
+    on_tpu = ctx["on_tpu"]
+    optimizer_closure = ctx.pop("optimizer_closure", do_nothing_optimizer_closure)
+    optimizer = trainer.optimizers[opt_idx]
+
+    is_lbfgs = isinstance(optimizer, torch.optim.LBFGS)
+    using_native_amp = trainer.amp_backend == AMPType.NATIVE
+    automatic_optimization = trainer.train_loop.automatic_optimization
+
+    # native amp + lbfgs is a no go right now
+    if using_native_amp and is_lbfgs:
+        raise MisconfigurationException(
+            'native PyTorch amp and lbfgs are not compatible.'
+            ' To request, please file a Github issue in PyTorch and tag @mcarilli')
+
+    # model hook
+    model_ref.optimizer_step(
+        epoch=trainer.current_epoch,
+        batch_idx=batch_idx,
+        optimizer=optimizer,
+        optimizer_idx=opt_idx,
+        optimizer_closure=optimizer_closure,
+        on_tpu=on_tpu,  # TPUAccelerator class sets this as True
+        using_native_amp=using_native_amp,
+        using_lbfgs=is_lbfgs,
+        *args,
+        **kwargs,
+    )
+
+
+class LightningPipeModule(nn.Module):
+    """
+        This class wraps Fairscale Pipe and PipeRCPWrapper class.
+
+        Args:
+            module: nn.Sequential
+                sequential model to be balanced among several gpus
+
+            balance: list of ints
+                list of number of layers in each partition.
+
+            checkpoint (str) = 'never'
+                when to enable checkpointing, one of ``'always'``,
+                ``'except_last'``, or ``'never'`` (default: ``'except_last'``)
+
+            balance_mode: str = "balance_by_size"
+                when balance is not provided, the model can be balanced either by size or time.
+                refer to balance description.
+
+            mode: PipeMode
+                the mode enables switching between Pipe and PipeRCPWrapper class
+    """
+    def __init__(self, module: nn.Sequential, balance: List[int],
+                 microbatches: int = 8, checkpoint='never', version: int = 1):
+        super().__init__()
+        assert version in [1, 2]
+        self._pipe_version = version
+        self.module = module
+        self.balance = balance
+        self.microbatches = microbatches
+        self.checkpoint = checkpoint
+        self._init_pipe()
+
+    def _init_pipe(self):
+        device = torch.device("cuda", torch_distrib.get_rank())
+        pipe_cls = Pipe if self._pipe_version == 1 else PipeRPCWrapper
```

Comment:
I suggest removing PipeRPCWrapper completely from the codebase to keep it simple!

Reply:
Yes, I just removed it.